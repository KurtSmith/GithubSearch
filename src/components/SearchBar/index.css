.btn-light, .btn{
    background-color: orange;
    outline: none;
  }
  .btn-check:focus+.btn-light, .btn-primary:focus {
    color: #fff;
    background-color: #D5A021 !important;
    border-color: #D5A021 !important;
    box-shadow: 0 0 0 0.2rem rgba(213,160,33, 0.5) !important;
  }
  .btn-primary:hover {
    background-color: #D5A021;
    border-color: #D5A021;
  }
  .btn-primary:hover {
    background-color: rgb(206, 136, 6);
  }
  .searchBox { 
  border: none;
  font-size: 20px;
  }
  
  .inputField { 
      margin: 5px;
  }