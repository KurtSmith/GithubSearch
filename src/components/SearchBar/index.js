import './index.css';
import React, { useContext, useState } from 'react';
import { useHistory } from 'react-router-dom';
import { InputGroup, FormControl, Button } from 'react-bootstrap';
import {updateTopics} from '../../context/actions'
import logo from '../../assets/search.svg';
import retrieveResults from '../../service';
import { GithubContext } from '../../context';

export default function SearchBar(props) {

    const [state, dispatch] = useContext(GithubContext);
    const [searchEnabled, setSearchEnabled] = useState(false);
    const history = useHistory();

    const getResults = (e) => {
        dispatch(updateTopics(state.currentTopic, state.currentTopic,[]));
        const onDetailPage = history.location.pathname !== '/';
        retrieveResults(dispatch, state.currentTopic, state.sortStars, state.languages, 1);
        if (onDetailPage)
            history.goBack();
    }
    // i never want to click a button so lets intercept the enter key and search
    const handleKeyDown = (event) => {
        if (event.key === 'Enter') {
            getResults();
        }
    }

    const inputChanged = (e) => {
        setSearchEnabled(e.target.value.length > 0);
        dispatch(updateTopics(e.target.value, state.lastTopic,state.results));
    }

    return (
        <InputGroup classlabel="mb-3">
            <FormControl
                data-testid="searchInput"
                placeholder="Enter repository name and click the search icon"
                aria-label="RepoSearch"
                aria-describedby="RepoSearch"
                value={state.currentTopic}
                onChange={inputChanged}
                onKeyDown={handleKeyDown}
                maxLength='50'
            />
            <Button onClick={getResults} data-testid="searchButton" className="btn btn-light" id="search" disabled={state.loading || !searchEnabled}>
                <img src={logo} alt="Search Repos" />
            </Button>
        </InputGroup>
    )
}