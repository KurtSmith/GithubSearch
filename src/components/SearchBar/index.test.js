import React from 'react';
import renderer from 'react-test-renderer';
import SearchBar from './index';
import ContactContextProvider from '../../context';

test('searchbar input length', () => {
  const component = renderer.create(
    <ContactContextProvider>
    <SearchBar/>
    </ContactContextProvider>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.children.find(item=> item.type==='input').props.maxLength).toEqual('50');

});