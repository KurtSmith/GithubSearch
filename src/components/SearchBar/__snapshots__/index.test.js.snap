// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchbar input length 1`] = `
<div
  className="input-group"
  classlabel="mb-3"
>
  <input
    aria-describedby="RepoSearch"
    aria-label="RepoSearch"
    className="form-control"
    data-testid="searchInput"
    maxLength="50"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Enter repository name and click the search icon"
    value=""
  />
  <button
    className="btn btn-light btn btn-primary"
    data-testid="searchButton"
    disabled={true}
    id="search"
    onClick={[Function]}
    type="button"
  >
    <img
      alt="Search Repos"
      src="search.svg"
    />
  </button>
</div>
`;
